ui/page/dcrdex/dcrdex_page.go

```diff
@@ -63,24 +75,81 @@ func (pg *DEXPage) OnNavigatedTo() {
 // eventually drawn on screen.
 // Part of the load.Page interface.
 func (pg *DEXPage) Layout(gtx C) D {
+	noMultipleWalMsg := pg.isMultipleAssetTypeWalletAvailable()
+	dexDisableMsg := pg.dexEnabled()
+	overlay := layout.Stacked(func(gtx C) D { return D{} })
+	c, _ := pg.dexCore.(*dexcore.DEXCore)
+	dexSet := c != nil
+	if noMultipleWalMsg != "" || dexDisableMsg != "" || !dexSet {
+		msg := dexDisableMsg
+		actionBtn := &pg.generalSettingsBtn
+		if msg == "" {
+			msg = noMultipleWalMsg
+			actionBtn = nil
+		}
+		if !dexSet {
+			msg = values.String(values.StrDEXInitErrorMsg)
+			actionBtn = nil
+		}
+
+		gtxCopy := gtx
+		overlay = layout.Stacked(func(gtx C) D {
+			return components.DisablePageWithOverlay(pg.Load, nil, gtxCopy, msg, actionBtn)
+		})
+		// Disable main page from receiving events.
+		gtx = gtx.Disabled()
+	}
 	return layout.Stack{}.Layout(gtx,
 		layout.Expanded(func(gtx C) D {
 			return cryptomaterial.LinearLayout{
 				Width:       cryptomaterial.MatchParent,
 				Height:      cryptomaterial.MatchParent,
 				Orientation: layout.Vertical,
 			}.Layout(gtx,
-				layout.Flexed(1, pg.CurrentPage().Layout),
+				layout.Flexed(1, func(gtx layout.Context) layout.Dimensions {
+					if !dexSet {
+						return D{}
+					}
+					return pg.CurrentPage().Layout(gtx)
+				}),
 			)
 		}),
```

Comment:
This can be defined as `pageUI` or something and if there's no need for an overlay, return this directly, else return this as part of a layout.Stack{}.